tests/gtest/avifrgbtoyuvtest.cc

```diff
@@ -764,19 +766,21 @@ TEST_P(YUVToRGBThreadingTest, TestIdentical) {
   yuv->matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_BT601;
   yuv->yuvRange = AVIF_RANGE_FULL;
 
-  // Fill YUV planes with random values.
+  // Fill YUVA planes with random values.
   srand(0xAABBCCDD);
   const int yuv_max = (1 << yuv_depth);
-  ASSERT_EQ(avifImageAllocatePlanes(yuv.get(), AVIF_PLANES_YUV),
+  ASSERT_EQ(avifImageAllocatePlanes(
+                yuv.get(), has_alpha ? AVIF_PLANES_ALL : AVIF_PLANES_YUV),
             AVIF_RESULT_OK);
-  avifPixelFormatInfo pixel_format_info;
-  avifGetPixelFormatInfo(yuv_format, &pixel_format_info);
-  const int plane_count = pixel_format_info.monochrome ? 1 : 3;
-  for (int plane = 0; plane < plane_count; ++plane) {
+  for (int plane = AVIF_CHAN_Y;
+       plane <= (has_alpha ? AVIF_CHAN_A : AVIF_CHAN_V); ++plane) {
+    const uint32_t plane_width = avifImagePlaneWidth(yuv.get(), plane);
+    if (plane_width == 0) continue;
```

Comment:
is this check needed given the loop index?

Reply:
yes the ordering of the planes is Y, U, V and A. this check is needed to skip U and V planes when the format is monochrome (monochrome could have alpha).

Reply:
Got it. I wasn't considering monochrome.